### PR TITLE
Improvements to invitedPage and fix referral call

### PR DESCRIPTION
**Pull request by saltrafael**
## Fixes

Issue Number: n/a

- Some strings were missing
- The whole component was repeating itself
- Resolve claim to get permanent url

### CODE REVIEW

**Comment by tzarebczan** [comment]
@infinite-persistence can you give this a look over code wise and I'll test in the am, please?

File: ui/component/invited/view.jsx [hunk 1]
```diff
@@ -7,229 +7,178 @@ import ClaimPreview from 'component/claimPreview';
...
+            navigate={`/$/${PAGES.AUTH}?redirect=/$/${PAGES.INVITE}/${referrerUri || ''}`}
+          />
+          <Button button="link" label={__('Skip')} onClick={handleDone} />
+        </div>
+      ),
+    });
+  } else {
+    Object.assign(cardProps, {
+      title: __('Welcome!'),
+      subtitle: referrerIsChannel ? __("We've followed your invitee for you. Check them out!") : __('Congrats!'),
```
[the remark this answers is not in the document]
Comment (infinite-persistence) [reply ui/component/invited/view.jsx:163]: But neither is correct, I think. The `null` claim case isn't handled ... maybe should just say "invalid link". But no need to change, just thinking out loud.

File: ui/component/invited/view.jsx [hunk 2]
```diff
@@ -7,229 +7,178 @@ import ClaimPreview from 'component/claimPreview';
...
     );
+
+    Object.assign(cardProps, {
+      title: referrerIsChannel
+        ? __('%channel_name% invites you to the party!', { channel_name: channelTitle })
+        : __("You're invited!"),
+      subtitle: referrerIsChannel ? (
+        <I18nMessage tokens={{ channel_name: channelTitle, signup_link: signUpButton, SITE_NAME }}>
+          %channel_name% is waiting for you on %SITE_NAME%. Create your account now.
+        </I18nMessage>
```
Comment (infinite-persistence) [inline ui/component/invited/view.jsx:143]: :small_red_triangle: Hm ... `channelTitle` is undefined for some channels (e.g. `@t2:2`), so the sentence could potentially be odd?

Fallback to claim name?

**Comment by tzarebczan** [comment]
let's get at least the permanent URL part fixed.

File: ui/component/invited/view.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (saltrafael) [reply ui/component/invited/view.jsx:163]: `null` claim now handled

File: ui/component/invited/view.jsx (hunk 2 above)
Comment (saltrafael) [reply ui/component/invited/view.jsx:143]: `channelTitle` comes from `getChannelTitleFromClaim` which already falls back to name in undefined cases

**Review by infinite-persistence — COMMENTED** [review]
General code-wise, it looks ok.

Not sure on logic/behavior-wise, because the new logic differences aren't obvious from the `diff`. 
We would need to list out all the permutations (valid/invalid inputs, rewards approved/not, etc.) and test them out. 
A more conservative fix would be preferred without a full test plan.

Will try to test all later today.

File: static/app-strings.json [hunk 3]
```diff
@@ -1001,7 +1001,6 @@
   "Create Account": "Create Account",
```
Comment (infinite-persistence) [inline static/app-strings.json:1001]: **Scenario:** new account, rewards not approved, email verified.

- [ ] I'm logged in, but subtitle still asking me to create an account (maybe legacy issue, at least the Button label 'Finish Account' makes sense).
- [ ] Pressing 'Finish Account' brings me to http://localhost:9090/$/invite/lbry://@dev-test and does nothing.  I think I should be trying to get my rewards approved first?

image

Comment (tzarebczan) [reply static/app-strings.json:1002]: There were some existing bugs with this process too...not something introduced.

File: static/app-strings.json [hunk 4]
```diff
@@ -900,6 +900,7 @@
   "Sorry about that. Contact %SITE_HELP_EMAIL% if you continue to have issues.": "Sorry about that. Contact %SITE_HELP_EMAIL% if you continue to have issues.",
```
Comment (infinite-persistence) [inline static/app-strings.json:900]: _(not sure if this is legacy or not, can't retry ... need to figure out how to reset account first)_

Scenario:  Account approved for rewards, but haven't claimed "Inviter Bonus" yet
Action:  http://localhost:9090/$/invite/@dev-test:8
Result:  I got the new "go get your Rewards approved" screen :x: .  I also see a bunch of toasts, and saw the inviter bonus.  After refresh, the screen became "you've already claim, we subscribed for you instead".

Maybe just a timing issue, pending fetch or something.

File: ui/page/rewardsVerify/index.jsx [hunk 5]
```diff
@@ -4,20 +4,8 @@ import UserVerify from 'component/userVerify';
...
-  const { user } = props;
+function RewardsVerifyPage() {
   const { goBack } = useHistory();
-  const rewardsApproved = user && user.is_reward_approved;
-
-  React.useEffect(() => {
-    if (rewardsApproved) {
-      goBack();
-    }
-  }, [rewardsApproved]);
```
Comment (infinite-persistence) [inline ui/page/rewardsVerify/index.jsx:20]: I always find the automatic `goBack()` in `/$/rewardsVerify` when already verified quite confusing.  This seems like a good direction.

:small_blue_diamond: A few things to consider/check:
- If I'm verified and I enter `/$/rewardsVerify` (say, from clicking an old notification, or due to some sequence), the page still allows you to perform the verification actions.  
    - Minor, though.  And not necessarily needed in this PR, but can gray out and say "already verified" in the future.
- I vaguely recall there is reliance on the old behavior in one of the flows ... can't recall if it's sign-up, sign-in, comment-when-no-credits, or something else.  (Can ignore this for now, I'll try to dig).

**Pull request merged**